api/v1alpha1/nodepool_types.go

```diff
@@ -286,6 +287,19 @@ type NodePoolPlatform struct {
 
 	// IBMCloud defines IBMCloud specific settings for components
 	IBMCloud *IBMCloudPlatformSpec `json:"ibmcloud,omitempty"`
+
+	// Kubevirt specifies the configuration used when operating on KubeVirt platform.
+	//
+	// +optional
+	// +immutable
+	Kubevirt *KubevirtNodePoolPlatform `json:"kubevirt,omitempty"`
+}
+
+// KubevirtNodePoolPlatform specifies the configuration of a NodePool when operating
+// on KubeVirt platform.
+type KubevirtNodePoolPlatform struct {
+	// NodeTemplate Spec contains the VirtualMachineInstance specification.
```

Comment:
I don't have strong opinions here but  curious if you have considered passing `kubevirtv1.VirtualMachineSpec` instead? have you considered embed the type? If not embedded have you considered naming NodeTemplate vs VirtualMachineTemplate vs Template?

Reply:
Also should `NodeTemplate` be required?

Reply:
ignore the required comment, it's already unless indicated otherwise.

Reply:
> I don't have strong opinions here but curious if you have considered passing kubevirtv1.VirtualMachineSpec instead (do you really need to expose the ObjectMeta)? have you considered embed the type? If not embedded have you considered naming NodeTemplate vs VirtualMachineTemplate vs Template?

I think being able to set annotations and labels is pretty important but also kubevirt specific. Therefore I think it makes sense to also have the objectmeta.

Reply:
Yes we need ObjectMeta on the KubeVirt VM itself that is independent of the NodePool. Even the Namespace is important to us.

Here's one reason why

Right now the kubevirt provider is assumed to be starting kubevirt VMs within the management cluster, that's not guaranteed to be the case in the future though. We're open to the possibility that the NodePool will be creating kubevirt machines in an external cluster. In this case, the namespace within the ObjectMeta on the VM here actually represents the namespace on the external cluster to launch the KubeVirt VM.